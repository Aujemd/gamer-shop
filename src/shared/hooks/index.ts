export { useStoredCart } from "./use-stored-cart";
