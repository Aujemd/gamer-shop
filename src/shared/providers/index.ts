export * from "./data-fetching-provider";
