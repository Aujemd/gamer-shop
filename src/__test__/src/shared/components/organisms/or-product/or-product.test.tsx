import { render, screen } from "@testing-library/react";

import { OrProduct } from "@/shared";

describe("orProduct component", () => {
    it("should render correctly", () => {
        const { asFragment } = render(
            <OrProduct>
                <li>Item 1</li>
                <li>Item 2</li>
            </OrProduct>,
        );

        expect(asFragment()).toMatchSnapshot();
    });

    it("renders children correctly", () => {
        render(
            <OrProduct data-testid="or-product-test-id">
                <li data-testid="child-1">Card 1</li>
                <li data-testid="child-2">Card 2</li>
            </OrProduct>,
        );

        expect(screen.getByTestId("child-1")).toHaveTextContent("Card 1");
        expect(screen.getByTestId("child-2")).toHaveTextContent("Card 2");
        expect(screen.getByTestId("or-product-test-id")).toBeInTheDocument();
    });

    it("applies custom className", () => {
        render(
            <OrProduct
                data-testid="or-product-test-id"
                className="custom-class"
            >
                <li>Card</li>
            </OrProduct>,
        );
        const ul = screen.getByTestId("or-product-test-id");
        expect(ul).toHaveClass("custom-class");
    });

    it("passes additional props to ul", () => {
        render(
            <OrProduct aria-label="product-list">
                <li>Card</li>
            </OrProduct>,
        );
        expect(screen.getByLabelText("product-list")).toBeInTheDocument();
    });
});
