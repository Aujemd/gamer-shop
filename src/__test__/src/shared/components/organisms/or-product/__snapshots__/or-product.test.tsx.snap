// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`orProduct component should render correctly 1`] = `
<DocumentFragment>
  <ul
    class="grid grid-cols-1 gap-6 py-8 px-6 md:grid-cols-2 lg:grid-cols-3 lg:gap-12 xl:pl-0 xl:pr-11 xl:py-12 items-stretch undefined"
  >
    <li>
      Item 1
    </li>
    <li>
      Item 2
    </li>
  </ul>
</DocumentFragment>
`;
